gov/politeia/proposals.go

```diff
@@ -91,373 +90,607 @@ func NewProposalsDB(politeiaURL, dbPath string) (*ProposalDB, error) {
+// data returned does not include ticket vote data.
+func (db *ProposalsDB) fetchProposalsData(tokens []string) ([]*pitypes.ProposalRecord, error) {
+	// Fetch record details for each token from the inventory.
+	recordDetails, err := db.fetchRecordDetails(tokens)
+	if err != nil {
+		return nil, err
 	}
 
-	return db.proposal("RefID", RefID)
-}
+	// Fetch comments count for each token from the inventory.
+	commentsCounts, err := db.fetchCommentsCounts(tokens)
+	if err != nil {
+		return nil, err
+	}
 
-// proposal runs the query with searchBy and searchTerm parameters provided and
-// returns the result.
-func (db *ProposalDB) proposal(searchBy, searchTerm string) (*pitypes.ProposalInfo, error) {
-	var pInfo pitypes.ProposalInfo
-	err := db.dbP.Select(q.Eq(searchBy, searchTerm)).Limit(1).First(&pInfo)
+	// Fetch vote summary for each token from the inventory.
+	voteSummaries, err := db.fetchTicketVoteSummaries(tokens)
 	if err != nil {
-		if !errors.Is(err, storm.ErrNotFound) {
-			log.Errorf("Failed to fetch data from Proposals DB: %v", err)
-		}
 		return nil, err
 	}
 
-	return &pInfo, nil
+	// Iterate through every record and feed data used by dcrdata
+	proposals := make([]*pitypes.ProposalRecord, 0, len(recordDetails))
+	for _, record := range recordDetails {
+		// Record data
+		proposal := &pitypes.ProposalRecord{
+			State:     record.State,
+			Status:    record.Status,
+			Version:   record.Version,
+			Timestamp: uint64(record.Timestamp),
+			Username:  record.Username,
+			Token:     record.CensorshipRecord.Token,
+		}
+
+		// Proposal metadata
+		pm, err := proposalMetadataDecode(record.Files)
+		if err != nil {
+			return nil, fmt.Errorf("proposalMetadataDecode err: %w", err)
+		}
+		proposal.Name = pm.Name
+
+		// User metadata
+		um, err := userMetadataDecode(record.Metadata)
+		if err != nil {
+			return nil, fmt.Errorf("userMetadataDecode err: %w", err)
+		}
+		proposal.UserID = um.UserID
+
+		// Comments count
+		commentsCount, ok := commentsCounts[proposal.Token]
+		if !ok {
+			log.Errorf("Comments count for proposal %v not returned by API",
+				proposal.Token)
+			continue
+		}
+		proposal.CommentsCount = int32(commentsCount)
+
+		// Vote summary data
+		summary, ok := voteSummaries[proposal.Token]
+		if !ok {
+			log.Errorf("Vote summary for proposal %v not returned by API",
+				proposal.Token)
+			continue
+		}
+		proposal.VoteStatus = summary.Status
+		proposal.VoteResults = summary.Results
+		proposal.EligibleTickets = summary.EligibleTickets
+		proposal.StartBlockHeight = summary.StartBlockHeight
+		proposal.EndBlockHeight = summary.EndBlockHeight
+		proposal.QuorumPercentage = summary.QuorumPercentage
+		proposal.PassPercentage = summary.PassPercentage
+
+		var totalVotes uint64
+		for _, v := range summary.Results {
+			totalVotes += v.Votes
+		}
+		proposal.TotalVotes = totalVotes
+
+		// Status change metadata
+		statusTimestamps, changeMsg, err :=
+			statusChangeMetadataDecode(record.Metadata)
```

Comment:
Only comments need to wrap strictly to 80.  This can go on 1 line as can a number of other fmt.Errorf statements with err on the next line.

The guidline for dcrdata style is that when it comes to code, columns are less important than readibility of the code, just keep is to less than a 100 or so columns, but no fixed limit.  Comments on the other hand I really like formatted consistently.  Setup your editor to wrap at 80 with some key combo.